File: src/main/java/com/bot/model/entities/Category.java
package com.bot.model.entities;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import java.util.List;

@Entity
@Getter
@Setter
@Table(name = "Category")
@NoArgsConstructor
public class Category {

    public final static int MAX_NAME_LENGTH = 25;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="band_id", nullable=false)
    private Band band;

    //can`t set cascade on remove cause band remove leads to category remove but all spendings must be saved
    @OneToMany(fetch = FetchType.LAZY, mappedBy = "category")
    private List<Product> products;
}

File: plunner-core/src/main/kotlin/com/planner/model/Category.kt
package com.planner.model

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table

/**
 * Entity represents category of products
 */
@Entity
@Table(name = "category")
class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id")
    var group: UserGroup? = null

    @OneToMany(mappedBy = "category", cascade = [CascadeType.REMOVE])
    var products: List<Product> = emptyList()

    @Column
    var name: String = "Category"
}
